### ISSUE TITLE
predict_instances hangs after all slices have been predicted

### ISSUE THREAD

**Issue by Xqua** [mpicbg-csbd/stardist#99]
First, sorry for all the issues I posted recently O.o, overall I really think StarDist is a really neat architecture! The results I am getting after training are really great !

Right now though, the `predict_instance` is not working anymore and I don't understand why. 
I have trained the model over 5 generations, at each generation, I gave the network 12 volumes (~600x500x500 cubes) as training, and 4 as validation. After 200 epochs (27s per epoch) the training stoped. I copied the weights to a new folder, and selected the next 12 and 4 volumes from the two sets. I reload the copied model for training and continued the training. 
This worked great with my dist_mae finishing at ~0.92 for training and ~0.96 for the validation set. 

After each training, I ran predicted the validation volumes to perform the IOU estimation of the pixel recovery. All good, about ~4min30 to predict a volume, so 18 min total. 

Now, I feel that the network is trained and ready. I create a new notebook and am about to feed my real validation dataset to see over 300 annotated volumes with Mamut the recovery rate of nucleis. 

Here is the code that I use for that:
```
from __future__ import print_function, unicode_literals, absolute_import, division
import sys, os
import shutil
import numpy as np
import matplotlib
matplotlib.rcParams["image.interpolation"] = None
import matplotlib.pyplot as plt
%matplotlib inline
%config InlineBackend.figure_format = 'retina'

from glob import glob
from tqdm import tqdm
from tifffile import imread
from csbdeep.utils import Path, normalize
import tensorflow as tf

from stardist import fill_label_holes, random_label_cmap, calculate_extents, gputools_available
from stardist import Rays_GoldenSpiral, relabel_image_stardist3D
from stardist.matching import matching, matching_dataset
from stardist.models import Config3D, StarDist3D, StarDistData3D

np.random.seed(42)
lbl_cmap = random_label_cmap()

# Deal with GPU growth : taken here https://www.tensorflow.org/guide/gpu
gpus = tf.config.experimental.list_physical_devices('GPU')
if gpus:
    try:
        # Currently, memory growth needs to be the same across GPUs
        for gpu in gpus:
            tf.config.experimental.set_memory_growth(gpu, True)
        logical_gpus = tf.config.experimental.list_logical_devices('GPU')
        print(len(gpus), "Physical GPUs,", len(logical_gpus), "Logical GPUs")
    except RuntimeError as e:
        # Memory growth must be set before GPUs have been initialized
        print(e)

import h5py

class Dataset:
    def __init__(self, path, channel='s17', level=0, min_frame=-1, max_frame=100000):
        self.H5 = h5py.File(path, 'r')
        self.__set_frames()
        self.channel = channel
        self.level = level
        self.min_frame = max(min(self.frames), min_frame)
        self.max_frame = min(max(self.frames), max_frame)
    
    def __len__(self):
        return len(self.frames)
    
    def __getitem__(self, idx):
        frame = self.frames[idx]
        im = self.__get_frame(frame)
        return im
    
    def __iter__(self):
        self.n = 0
        return self

    def __next__(self):
        if self.n <= self.__len__():
            im = self.__getitem__(self.n)
            self.n += 1
            return im
        else:
            raise StopIteration
    
    def __set_frames(self):
        frames = []
        for key in self.H5.keys():
            if 't' in key:
                frames.append(int(key.replace('t','')))
        self.frames = sorted(frames)
    
    def __get_frame(self, frame):
        T = "t{0:05d}".format(frame)
        path = "{}/{}/{}/cells".format(T, self.channel, self.level)
        im = self.H5.get(path)[()]
        return im

generator = Dataset('path/to/Bro1-fused2.h5', level=0)

# testing on one frame
im = generator[100]
model_name = "stardist_lightsheet_5"
model = StarDist3D(config=None, name=model_name, basedir='models')
#Loading network weights from 'weights_best.h5'.
#Couldn't load thresholds from 'thresholds.json', using default values. (Call 'optimize_thresholds' to change that.)
#Using default values: prob_thresh=0.5, nms_thresh=0.4.
print(model._guess_n_tiles(im))
# (14, 8, 7)
pred = model.predict_instances(im, n_tiles=model._guess_n_tiles(im), show_tile_progress=True)[0]
# 100%|██████████| 784/784 [04:18<00:00,  3.03it/s]
```

And then it hangs, for hours, with 16 CPUs running at 100% and 32Gb of RAM used. The GPU however is at 10% usage. 
This sounds very similar to what this #88 user described at first. 

I'm confused because AFTER training the model worked fine to predict images. But I guess upon reloading it doesn't ?

**Comment by maweigert** [comment]
Did you normalize the image the same way as during training ?

**Comment by Xqua** [comment]
I was coming back to say that I did not normalize the images -_-' !!!
I just realized and tries with normalized images and its working again ...

My bad, thanks for the anwser !!
